std/parallelism.d

```diff
@@ -4303,243 +4321,311 @@ version(parallelismStressTest)
-            assert(mySum == 999 * 1000 / 2);
+    ulong sumSquares = 0;
+    foreach (elem; numbers)
+    {
+        sumSquares += elem * elem;
+    }
 
-            auto mySumParallel = poolInstance.reduce!"a + b"(numbers);
-            assert(mySum == mySumParallel);
-            stderr.writeln("Done sums.");
+    uint mySum = sumFuture.spinForce();
+    assert(mySum == 999 * 1000 / 2);
 
-            auto myTask = task(
-            {
-                synchronized writeln("Our lives are parallel...Our lives are parallel.");
-            });
-            poolInstance.put(myTask);
+    auto mySumParallel = poolInstance.reduce!"a + b"(numbers);
+    assert(mySum == mySumParallel);
+    version(parallelismStressTest) stderr.writeln("Done sums.");
 
-            auto nestedOuter = "abcd";
-            auto nestedInner =  iota(0, 10, 2);
+    version(parallelismStressTest)
+    {
+        auto myTask = task(
+        {
+            synchronized writeln("Our lives are parallel...Our lives are parallel.");
+        });
+    }
+    else
+    {
+        auto myTask = task(
+        {
+            synchronized {}
+        });
+    }
+    poolInstance.put(myTask);
+
+    auto nestedOuter = "abcd";
+    auto nestedInner =  iota(0, 10, 2);
 
-            foreach (i, letter; poolInstance.parallel(nestedOuter, 1))
+    version(parallelismStressTest)
+    {
+        foreach (i, letter; poolInstance.parallel(nestedOuter, 1))
+        {
+            foreach (j, number; poolInstance.parallel(nestedInner, 1))
             {
-                foreach (j, number; poolInstance.parallel(nestedInner, 1))
-                {
-                    synchronized writeln(i, ": ", letter, "  ", j, ": ", number);
-                }
+                synchronized writeln(i, ": ", letter, "  ", j, ": ", number);
+            }
+        }
+    }
+    else
+    {
+        foreach (i, letter; poolInstance.parallel(nestedOuter, 1))
+        {
+            foreach (j, number; poolInstance.parallel(nestedInner, 1))
+            {
+                synchronized {}
             }
-
-            poolInstance.stop();
         }
-
-        assert(attempt == 10);
-        writeln("Press enter to go to next round of unittests.");
-        readln();
     }
 
-    // These unittests are intended more for actual testing and not so much
-    // as examples.
+    poolInstance.stop();
+}
+
+version(parallelismStressTest)
+{
     unittest
     {
         foreach (attempt; 0..10)
         foreach (poolSize; [0, 4])
         {
-            poolInstance = new TaskPool(poolSize);
+            stressTest2(poolSize);
+        }
+    }
+}
+else
+{
+    unittest
+    {
+        stressTest2(1);
+    }
+}
 
-            // Test indexing.
-            stderr.writeln("Creator Raw Index:  ", poolInstance.threadIndex);
-            assert(poolInstance.workerIndex() == 0);
+// These unittests are intended more for actual testing and not so much
+// as examples.
+void stressTest2(int poolSize)
```

Comment:
I created this function to have runnable one or multiple times. Eventually, this will be private. But this is not the problem here. Please keep the nitpicking to a minimum for now.